x-pack/plugins/infra/public/containers/logs/log_summary/with_summary.ts

```diff
@@ -22,15 +22,18 @@ export const WithSummary = ({
     end: number | null;
   }>;
 }) => {
-  const { intervalSize } = useContext(LogViewConfiguration.Context);
   const { sourceId } = useContext(Source.Context);
   const { filterQuery } = useContext(LogFilterState.Context);
-  const { visibleMidpointTime } = useContext(LogPositionState.Context);
+  const { startTimestamp, endTimestamp } = useContext(LogPositionState.Context);
+
+  // Keep it reasonably updated for the `now` case, but don't reload all the time when the user scrolls
+  const throttledStartTimestamp = useThrottle(startTimestamp, 3000);
+  const throttledEndTimestamp = useThrottle(endTimestamp, 3000);
```

Comment:
I did that in the beginning, but I didn't feel comfortable with the lack of determinism. I think the `useLogSummary` should always give a valid output for the given input. If the throttle inside the hook it might return outdated values for different inputs.

I think the responsibility of throttling should be in the consumer of `useLogSummary`.

Reply:
yes, that makes sense to me